internal/jsonfieldnames/extract.go

```diff
@@ -0,0 +1,98 @@
+// Package jsonfieldnames contains a function to extract JSON field names from a value.
+//
+// In addition to their fixed, hardcoded members, business objects for energy marshal additional values into Geschaeftsobjekt.ExtensionData.
+// As Go's standard library JSON package has no option for a kitchen sink field accepting all remaining JSON fields,
+// there are two passes of unmarshaling: Once into the struct, once into a generic map. To avoid duplicates in the struct and the map,
+// fields that can be found in the struct are removed from the map. This package provides a mechanism to extract the names of
+// these fields in a generic way.
+package jsonfieldnames
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// Extract extracts JSON field names from a struct value or a pointer to a struct value. If the value passed to Extract is nil,
+// is not a struct value or a pointer to a struct value, an error is returned instead.
```

Comment:
```suggestion
// Extract extracts JSON field names from a struct value or a pointer to a struct value. If the value passed to Extract is either nil
// or is not a struct value or a pointer to a struct value, an error is returned instead.
```
iorgendwie hatte ich nach dem komma den "dann" satz erwartet statt eines weiteren cases, der zum error führt. nur vom lesefluss her.